File: 1924.cpp
#include<stdio.h>

int main() {
	int x, y, sum = 0;
	char month[12] = { 31,28,31,30,31,30,31,31,30,31,30,31 };
	char day[7][4] = { { "SUN" },{ "MON" },{ "TUE" },{ "WED" },{ "THU" },{ "FRI" },{ "SAT" } };
	scanf("%d%d", &x, &y);

	for (int i = 1; i < x; i++) {
		sum += month[i - 1];
	}
	sum += y;
	printf("%s", day[sum % 7]);
	return 0;
}

File: BOJ/python/1924.py
import sys

input = sys.stdin.readline

month = [-1, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
day = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"]
x, y = map(int, input().rstrip("\n").split(" "))
total = 0
for i in range(1, x):  # x월 까지 며칠이 걸리는지 더해준다.
    total += month[i]

if total == 0:  # x가 1월일 경우 '일'만 고려하면 된다.
    print(day[(y - 1) % 7])
else:  # x가 1월이 아닐 경우 total에 y일까지 며칠이 걸리는지 더해준다.
    total += y - 1
    print(day[total % 7])
